ui/components/component-library/popover/popover.tsx

```diff
@@ -92,13 +94,31 @@ export const Popover: PopoverComponent = React.forwardRef(
       }
     };
 
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        isOpen &&
+        popoverRef.current &&
+        !popoverRef.current.contains(event.target as Node)
+      ) {
+        if (onClickOutside) {
+          onClickOutside();
+        }
+      }
+    };
+
     useEffect(() => {
       document.addEventListener('keydown', handleEscKey);
+      if (isOpen) {
+        document.addEventListener('click', handleClickOutside);
+      } else {
+        document.removeEventListener('click', handleClickOutside);
+      }
```

Comment:
Since we're already adding on the 'keydown' event listener for the escape key here, what if we just go ahead and add the 'click' event listener too, regardless of whether isOpen is true or not? Might make things a bit simpler and more consistent, wdyt?